cmd/cli/plugin/package/package_available_get.go

```diff
@@ -156,3 +169,50 @@ func getValuesSchemaForPackage(namespace, name, version string, kc kappclient.Cl
 
 	return nil
 }
+
+func generateDefaultValuesForPackage(pkg *kapppkg.Package, valuesFile ...*os.File) error {
+	if pkg == nil {
+		// should not happen
+		return errors.New("pkg is nil")
+	}
+
+	if len(pkg.Spec.ValuesSchema.OpenAPIv3.Raw) == 0 {
+		log.Warningf("package '%s/%s' does not have any user configurable values in the '%s' namespace", pkgName, pkgVersion, packageAvailableOp.Namespace)
+		return nil
+	}
+
+	fileNamePrefix := ""
+	pkgNameTokens := strings.Split(pkgName, ".")
+	if len(pkgNameTokens) >= 1 {
+		fileNamePrefix = fmt.Sprintf("%s-", pkgNameTokens[0])
+	}
+	valuesFileName := fmt.Sprintf("%sdefault-values.yaml", fileNamePrefix)
+	var valuesFileToUse *os.File
+
+	if len(valuesFile) > 0 && valuesFile[0] != nil {
+		// caller is responsible for closing file
+		valuesFileToUse = valuesFile[0]
```

Comment:
Question: Why we need this `valuesFile` param, I don't see it being used. Is it just for testing purposes?

Reply:
Yeah valuesFile is not required. Used by unit test to assert output of the file since there is no easy way to mock os.File.